From f72a30e924bd985670e5594e612ab130d352aea8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 14 Jan 2021 23:35:33 +0200
Subject: [PATCH] [SW-20719] gaudi: report memory clear capability

After the HBM init has succeeded, report the HBM memory is clear.

Change-Id: Ia442da503e29bb8fdd1359d3221fe839c9d09aee
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 78dfce1..e055903 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -24,6 +24,7 @@
 #include <sha256.h>
 #include <boot_desc.h>
 #include <boot_if.h>
+#include <hl_fw_if.h>
 
 #include "common.h"
 
@@ -217,9 +218,14 @@
 
 void hl_ram_init(void)
 {
+	int err;
+
 	preboot_puts_raw("HBM init ");
-	if (gaudi_hbm_init())
+	err = gaudi_hbm_init();
+	if (err) {
 		preboot_puts("failed!\n");
-	else
+	} else {
+		preboot_set_enabled_cap(HL_FW_EN_CAPS_MEM_CLR);
 		preboot_puts("finished\n");
+	}
 }
